### ISSUE TITLE
Nullable warnings should be suppressed for `Assert.Catch()`

### ISSUE THREAD

**Issue by aayjaychan** [nunit/nunit.analyzers#345]
Nullable warnings for `Assert.Throw()` are now suppressed (#333). The same suppression should also be applied to `Assert.Catch()`.

The code:
```c#
Exception exception = Assert.Catch<Exception>(() => throw new InvalidOperationException());
```
current generates warning `CS8602 Dereference of a possibly null reference`.

**Comment by manfred-brands** [comment]
And whilst we add it, also `Assert.ThrowAsync` and `Assert.CatchAsync`.
@mikkelbu Please assign to me.

**Comment by mikkelbu** [comment]
Thanks @manfred-brands.

**Comment by manfred-brands** [comment]
Added fix to #343
